### PR TITLE
Fix documentations that were incorrect or missing

### PR DESCRIPTION
**Pull request by KingDGrizzle**
**Please describe the changes this PR makes and why it should be merged:**
Fixes some documentation "issues"
Check the commit for everything that was fixed

**Semantic versioning classification:**  
- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [x] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by iCrawl** [comment]
Can you take a full look at the docs and add anything you spot here. I would like to avoid countless doc PRs with just one change.

**Comment by KingDGrizzle** [comment]
I shall look then, I'll give you the ok sign when I looked everywhere. This bases on master wikis

**Comment by KingDGrizzle** [comment]
In ClientUserSettings, `this.user` isn't documented.. Intended or?

**Comment by devsnek** [comment]
intended

**Comment by SpaceEEC** [comment]
Some things I noticed:
- [ClientDataResolver#resolveImage](https://github.com/hydrabolt/discord.js/blob/master/src/client/ClientDataResolver.js#L174) is nullable.
- [Attachment#_attach](https://github.com/hydrabolt/discord.js/blob/master/src/structures/Attachment.js#L66) is missing a `@returns` notation.
- [ClientUserChannelOverride#patch](https://github.com/hydrabolt/discord.js/blob/master/src/structures/ClientUserChannelOverride.js#L14) is missing a `@returns` notation and should be `@private` like other patch methods.
- [ClientUserGuildSettings#patch](https://github.com/hydrabolt/discord.js/blob/master/src/structures/ClientUserGuildSettings.js#L28) ^
- [ClientUserSettings#patch](https://github.com/hydrabolt/discord.js/blob/master/src/structures/ClientUserSettings.js#L16) ^
- [ClientUserSettings#setGuildPosition](https://github.com/hydrabolt/discord.js/blob/master/src/structures/ClientUserSettings.js#L40) is missing a docstring for the method itself.
- [Guild#allowDMs](https://github.com/hydrabolt/discord.js/blob/master/src/structures/Guild.js#L785) is missing a warning that this feature is for user accounts only.

**Comment by KingDGrizzle** [comment]
That's gonna help immensely! Thanks for that

**Comment by KingDGrizzle** [comment]
- Attachment#_attach is missing a return statement, hence why its missing a `@returns`

**Comment by KingDGrizzle** [comment]
Uhh.

Shouldn't VoiceConnection#sendVoiceStateUpdate, setSessionID and setTokenAndEndpoint be private?

**Comment by KingDGrizzle** [comment]
Webhook#edit: options isn't optional but its params are?

File: src/structures/ClientApplication.js [hunk 1]
```diff
@@ -196,6 +196,9 @@ class ClientApplication extends Base {
   /**
    * When concatenated with a string, this automatically concatenates the app name rather than the app object.
    * @returns {string}
+   * @example
```
Comment (iCrawl) [inline src/structures/ClientApplication.js]: This seems immensely pointless.

**Comment by KingDGrizzle** [comment]
I ment to say remove example from ClientApplication#toString :P

**Comment by KingDGrizzle** [comment]
I am done here, except for the following things (condensing at its finest):

- Attachment#_attach not returning anything
- VoiceConnection#sendVoiceStateUpdate, setSessionID and setTokenAndEndpoint should be private?
- Webhook#edit: options isn't optional but its params are?
- ClientUserChannelOverride#patch, ClientUserGuildSettings#patch and ClientUserSettings#patch don't return anything

**Comment by SpaceEEC** [comment]
- Looks like we only annotate `@returns {void}` when eslint would complain, so I guess it's not needed here.
- Those methods of `VoiceConnection` look indeed private to me.
- `Webhook#edit`'s options parameter being not optional looks okay to me, as you want to actually edit something when calling that method.

**Comment by KingDGrizzle** [comment]
Perfect. Gonna commit the VoiceConnections one and it should be done!

**Comment by KingDGrizzle** [comment]
Actually, ClientUserSettings#update, is that supposed to be public?

**Comment by SpaceEEC** [comment]
I am not sure whether it is supposed to be public, since there is no conversion from camelCase to snek_case it's unlikely.
On another note, the second parameter's type `value` is not valid and should probably be `*`.

**Comment by KingDGrizzle** [comment]
From what I see, ClientUserSettings calls update manually with the snek_case, so the user doesn't have to. It probably is supposed to be private.
https://github.com/hydrabolt/discord.js/blob/3c532f16bc48175597a7bbed4fbba4637812184d/src/structures/ClientUserSettings.js#L48

**Comment by SpaceEEC** [comment]
Just saw that [`GuildEditData`](https://discord.js.org/#/docs/main/master/typedef/GuildEditData) is missing the `systemChannel` property (`ChannelResolvable`). See [here](https://github.com/hydrabolt/discord.js/blob/master/src/structures/Guild.js#L604).

**Comment by KingDGrizzle** [comment]
Wow I can spell can't I? `systemChannel` not `Ssystemchannel`

**Pull request merged**